packages/beacon-node/src/api/impl/beacon/pool/index.ts

```diff
@@ -91,7 +92,7 @@ export function getBeaconPoolApi({
               return;
             }
 
-            errors.push(e as Error);
+            errors.push({index: i, error: e as Error});
```

Comment:
Since this is not a plain array of errors anymore, we could consider just tracking failures here directly as we log the errors (incl. stack trace) separately anyways

```suggestion
            failures.push({index: i, message: (e as Error).message});
```